### DISCUSSION TITLE
on: check_suite never triggers, why?

### DISCUSSION THREAD

**Discussion by Xotl** [community/community#26169] [Actions]
<p>I’m trying to create a <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#check-suite-event-check_suite" rel="nofollow noopener">workflow that triggers after a successful check suite</a>, so i created a test workflow just to have a look at the event data, but my workflow never triggers and i don’t know why.</p>
<p>Here’s the workflow that i used:</p>
<pre><code>name: 'Check suite'
on:
  check_suite:
    types: [rerequested, completed]

jobs:
  something:
    name: 'some job'
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Event info
        run: echo "${{toJson(github)}}"
</code></pre>
<p>Any ideas why it never triggers? :confused:</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>According to the official document <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdeveloper.github.com%2Fv3%2Fchecks%2Fsuites%2F&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7Ce716e334df294253446d08d7927df01e%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637138941903925807&amp;sdata=OipMz2xNimu2z4GzmznTJFjSXwr%2BZkT3ZoQgTDgityE%3D&amp;reserved=0" rel="nofollow noopener">https://developer.github.com/v3/checks/suites/</a>:</p>
<p>Write permission for the Checks API is only available to GitHub Apps. OAuth Apps and authenticated users can view check runs and check suites, but they are not able to create them. If you aren’t building a GitHub App, you might be interested in the <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdeveloper.github.com%2Fv3%2Frepos%2Fstatuses%2F&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7Ce716e334df294253446d08d7927df01e%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637138941903935801&amp;sdata=JGQ%2BypfOlm3Dv8Bh0kez0B1sHgWicjwER%2Fs%2BmiQYjro%3D&amp;reserved=0" rel="nofollow noopener">Statuses API</a>.</p>
<p>There is previous ticket related to check suite event: <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.community%2Ft5%2FGitHub-Actions%2Fon-check-suite-never-gets-called%2Ftd-p%2F38955&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7Ce716e334df294253446d08d7927df01e%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637138941903935801&amp;sdata=po5Hj8i7chtqaWSH0lGVENqnyINB9holqhpAtfQ%2BUrQ%3D&amp;reserved=0" rel="nofollow noopener">https://github.community/t5/GitHub-Actions/on-check-suite-never-gets-called/td-p/38955</a>  TravisCI didn’t create check-suite and check-runs , so on: check_suite workflow was not triggered.</p>
<p>How did you create check suite event? Using Github App?</p>

**Comment by Xotl** [comment]
<p>I didn´t create a Github App, just wanted to create a workflow that listens to check_suite events.</p>
<p>Based on what you mentioned, only the check suites that are created by Github Apps are the ones that i can listen in my workflow, did i get that correctly?.</p>

**Comment by Yanjingzhu** [comment]
<p>Yeah, you are right. </p>
<p>Only GitHub Apps can create check suites and check runs. We have a guide for building a GitHub App here:</p>
<p><a href="https://developer.github.com/apps/building-github-apps/" rel="nofollow noopener">https://developer.github.com/apps/building-github-apps/</a></p>
<p>Once the GitHub App is created, it must be installed to at least one repository. From there, you can use an <a href="https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-an-installation" rel="nofollow noopener">installation access token</a> to authenticate with the GitHub API and create that check suite or check run:</p>
<p><a href="https://developer.github.com/v3/checks/suites/#create-a-check-suite" rel="nofollow noopener">https://developer.github.com/v3/checks/suites/#create-a-check-suite</a></p>
<p><a href="https://developer.github.com/v3/checks/runs/#create-a-check-run" rel="nofollow noopener">https://developer.github.com/v3/checks/runs/#create-a-check-run</a></p>
<p>Example:</p>
<pre><code>curl -v -X POST -H "Accept: application/vnd.github.antiope-preview+json" -H "Authorization: token $INSTALLATION_TOKEN" https://api.github.com/repos/:owner/:repo/check-suites
</code></pre>

**Comment by Xotl** [comment]
<p>i think i will use the status event, thanks for replying.</p>

**Comment by Xotl** [comment]
<p>hi <a class="mention" href="/yanjingzhu">@yanjingzhu</a>, i tried to listen to status event, but won’t trigger anyways.</p>
<p>Here’s the workflow:</p>
<pre><code>name: 'Check status data'
on: status

env:
  THE_GITHUB_CONTEXT: ${{ toJson(github) }}

jobs:
  something:
    name: 'some job'
    runs-on: ubuntu-latest

    steps:
      - name: Print context
        run: echo "Wow, such debugging ${{ toJson(github.event) }}"
</code></pre>
<p>And i’m posting an status with this request</p>
<pre><code>curl -L -X POST -H "Content-Type: application/json" -H "Authorization: token ***" -d '{ "state": "pending", "target_url": "https://example.com/build/status", "description": "The build succeeded!", "context": "continuous-integration/jenkins" }' "https://api.github.com/repos/Xotl/cool-github-releases/statuses/e58c04b8fbb610e2397d3c28b8b34634a30016e3"
</code></pre>
<p>The API responds with a 200 HTTP code &amp; i’m able to see my custom status on that commit:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3ff5dd8fc1fc07254b0132ce7964a5ad4e10eb8d.png" data-download-href="/uploads/short-url/97OQo8sDQBNz7qJDpSKKwOBAFVH.png?dl=1" title="GH status post.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7025727/181095908-3274209a-fa8f-405a-929b-355f1a8e2c93.png" alt="GH status post.png" data-base62-sha1="97OQo8sDQBNz7qJDpSKKwOBAFVH" width="690" height="56" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/3ff5dd8fc1fc07254b0132ce7964a5ad4e10eb8d_2_690x56.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/3ff5dd8fc1fc07254b0132ce7964a5ad4e10eb8d_2_1035x84.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/3ff5dd8fc1fc07254b0132ce7964a5ad4e10eb8d_2_1380x112.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3ff5dd8fc1fc07254b0132ce7964a5ad4e10eb8d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">GH status post.png</span><span class="informations">1456×120 11.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>yet, it never trigger my workflow. is this a bug? :thinking:</p>

**Comment by Yanjingzhu** [comment]
<p>I tested on my side. Create status API could trigger a worlflow with on: status. </p>
<p>Please see my example: </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/dbd35f2e23dc103d3d1e5f096300881efe7806a1.png" data-download-href="/uploads/short-url/vmFqxik7D4mM1Ne03p2Hvtf0bER.png?dl=1" title="Status API.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095909-06072999-743f-47a3-8fde-65820f308622.png" alt="Status API.png" data-base62-sha1="vmFqxik7D4mM1Ne03p2Hvtf0bER" width="690" height="167" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/dbd35f2e23dc103d3d1e5f096300881efe7806a1_2_690x167.png, https://aws1.discourse-cdn.com/github/optimized/2X/d/dbd35f2e23dc103d3d1e5f096300881efe7806a1_2_1035x250.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/d/dbd35f2e23dc103d3d1e5f096300881efe7806a1_2_1380x334.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/dbd35f2e23dc103d3d1e5f096300881efe7806a1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Status API.png</span><span class="informations">1812×440 108 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4ae90835a99c380a5ebb618df9b03f0f8b4ae584.png" data-download-href="/uploads/short-url/aGGBj0gB66sBoNdmhUDJfzjZ2Cw.png?dl=1" title="on status.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095910-46b96c89-ec6c-4cf0-9e93-5933913461ee.png" alt="on status.png" data-base62-sha1="aGGBj0gB66sBoNdmhUDJfzjZ2Cw" width="690" height="225" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4ae90835a99c380a5ebb618df9b03f0f8b4ae584_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">on status.png</span><span class="informations">835×273 16.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>When I try to create a status using Postman, the response status is 201 Created. Could you please try to create a status again? </p>

**Comment by Xotl** [comment]
<p>Tested again and still not working, the <strong>on: status</strong> never triggers my workflow. I see that you did some test on master, i’m trying to trigger a workflow on a branch that it’s not my default. Does that could be the problem? i don’t see anything related on the documentaion.</p>
<p><strong>EDIT: </strong> It only works on master branch. Tested it and it’s working now.</p>

**Comment by Yanjingzhu** [comment]
<p>Hi Xotl, </p>
<p>Yeah, you are right, status event only work on master branch. You need to have a workflow with on: status in master branch. Sorry for not telling you in advance。 </p>

**Comment by jrencz** [comment]
<p>Hi Yanjingzhu, </p>
<blockquote>
<p>You need to have a workflow with on: status in master branch</p>
</blockquote>
<p>is it exactly “master” or “the main branch as set in repo settings”?</p>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/jrencz">@jrencz</a> It is the default/main branch you set in your repo settings. Sorry for the careless. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4aefc0e35f1ed8a19bea1eae895a33681933d4c0.png" data-download-href="/uploads/short-url/aGV08EOHMWiNWvwcPmi4SHT4Yi4.png?dl=1" title="default branch.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095912-2956792f-0681-4daf-8a15-c926a7ed812f.png" alt="default branch.png" data-base62-sha1="aGV08EOHMWiNWvwcPmi4SHT4Yi4" width="690" height="399" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/4aefc0e35f1ed8a19bea1eae895a33681933d4c0_2_690x399.png, https://aws1.discourse-cdn.com/github/original/2X/4/4aefc0e35f1ed8a19bea1eae895a33681933d4c0.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/4/4aefc0e35f1ed8a19bea1eae895a33681933d4c0.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4aefc0e35f1ed8a19bea1eae895a33681933d4c0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">default branch.png</span><span class="informations">987×571 31.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
